[BUG] Redundant unique constraints cause postgres db error

**Describe the bug**

When creating the tables on postgres, I get the error:

```
E       sqlalchemy.exc.ProgrammingError: (psycopg2.errors.DuplicateTable) relation "uq_ml_model_versions_resource_snapshot_id" already exists
E
E       [SQL:
E       CREATE TABLE ml_model_versions (
E               resource_snapshot_id BIGINT NOT NULL,
E               resource_id BIGINT NOT NULL,
E               artifact_resource_snapshot_id BIGINT NOT NULL,
E               version_number BIGINT NOT NULL,
E               CONSTRAINT pk_ml_model_versions PRIMARY KEY (resource_snapshot_id),
E               CONSTRAINT fk_ml_model_versions_resource_snapshot_id_resource_snapshots FOREIGN KEY(resource_snapshot_id, resource_id) REFERENCES resource_snapshots (resource_snapshot_id, resource_id),
E               CONSTRAINT uq_ml_model_versions_resource_snapshot_id UNIQUE (resource_snapshot_id, resource_id, artifact_resource_snapshot_id, version_number),
E               CONSTRAINT uq_ml_model_versions_resource_snapshot_id UNIQUE (resource_snapshot_id, resource_id),
E               CONSTRAINT fk_ml_model_versions_artifact_resource_snapshot_id_artifacts FOREIGN KEY(artifact_resource_snapshot_id) REFERENCES artifacts (resource_snapshot_id)
E       )
E
E       ]
```

**To Reproduce**

Try to create all the tables using the postgres database.

**Expected behavior**

Expect all tables created, no errors.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - DB: Postgres 16.2

**Explanation**

Notice that the `uq_ml_model_versions_resource_snapshot_id` constraint is listed twice.  There is a two-column unique constraint and a four-column unique constraint, where the four-column constraint contains all the columns of the two-column constraint.  That makes the two constraints redundant.  If the two-column is satisfied, then the four-column is satisfied (and not vice versa).  So the two-column is "stricter" in that sense.  So you could drop the four-column constraint with no effective change.

--- a/src/dioptra/restapi/db/alembic/versions/fd786b5377d6_add_ml_model_versions_resource.py
+++ b/src/dioptra/restapi/db/alembic/versions/fd786b5377d6_add_ml_model_versions_resource.py
@@ -208,16 +208,6 @@ def upgrade():
         sa.UniqueConstraint(
             "resource_snapshot_id",
             "resource_id",
-            "artifact_resource_snapshot_id",
-            "version_number",
-            name=op.f(
-                "uq_ml_model_versions_resource_snapshot_id_artifact_resource_snapshot"
-                "_id_version_number"
-            ),
-        ),
-        sa.UniqueConstraint(
-            "resource_snapshot_id",
-            "resource_id",
             name=op.f("uq_ml_model_versions_resource_snapshot_id"),
         ),
     )

--- a/src/dioptra/restapi/db/models/ml_models.py
+++ b/src/dioptra/restapi/db/models/ml_models.py
@@ -113,12 +113,6 @@ class MlModelVersion(ResourceSnapshot):
                 "resource_snapshots.resource_id",
             ],
         ),
-        UniqueConstraint(
-            "resource_snapshot_id",
-            "resource_id",
-            "artifact_resource_snapshot_id",
-            "version_number",
-        ),
         UniqueConstraint("resource_snapshot_id", "resource_id"),
     )
     __mapper_args__ = {

